torch/distributed/fsdp/_state_dict_utils.py

```diff
@@ -338,6 +331,7 @@ def _full_pre_load_state_dict_hook(
     state_dict: Dict[str, Any],
     prefix: str,
 ) -> None:
+    _lazy_init(fsdp_state, module)
```

Comment:
Could you remind me: Is there any functional change that happens by moving the `_lazy_init()` call into the pre-hook?

In my current understanding, there is no change (since `_lazy_init()` is a no-op if already initialized and since the hooks will run top-down as desired).

Reply:
Yes, if `pre_state_dict_hook` is called from root to leaf, there is no functional change.